### PR TITLE
Integrate datasets crud endpoints

### PR DESCRIPTION
**Pull request by frascuchon**
This PR integrates new DB logic and policies with the current implementation.

- The main changes are applied to the dataset service class
- The `auth.get_current_user` returns a `models.User` instance
- The ONLY change in endpoints is using this new `auth.get_current_user` method

In a separate PR the pydantic User class will be cleaned and normalized

### CODE REVIEW

File: src/argilla/server/apis/v0/handlers/datasets.py [hunk 1]
```diff
@@ -72,7 +67,7 @@ async def create_dataset(
     request: CreateDatasetRequest = Body(..., description="The request dataset info"),
     ws_params: CommonTaskHandlerDependencies = Depends(),
     datasets: DatasetsService = Depends(DatasetsService.get_instance),
-    user: User = Security(auth.get_user, scopes=["create:datasets"]),
+    user: User = Security(auth.get_current_user),
```
Comment (jfcalvo) [inline src/argilla/server/apis/v0/handlers/datasets.py]:
```suggestion
    current_user: User = Security(auth.get_current_user),
```

File: src/argilla/server/apis/v0/handlers/datasets.py [hunk 2]
```diff
@@ -72,7 +67,7 @@ async def create_dataset(
...
 ) -> Dataset:
     request.workspace = request.workspace or ws_params.workspace
     dataset = datasets.create_dataset(user=user, dataset=request)
```
Comment (jfcalvo) [inline src/argilla/server/apis/v0/handlers/datasets.py]:
```suggestion
    dataset = datasets.create_dataset(user=current_user, dataset=request)
```

File: src/argilla/server/apis/v0/handlers/text_classification_dataset_settings.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ async def get_dataset_settings(
         name: str = DATASET_NAME_PATH_PARAM,
         ws_params: CommonTaskHandlerDependencies = Depends(),
         datasets: DatasetsService = Depends(DatasetsService.get_instance),
-        user: User = Security(auth.get_user, scopes=["read:dataset.settings"]),
+        user: User = Security(auth.get_current_user),
```
Comment (jfcalvo) [inline src/argilla/server/apis/v0/handlers/text_classification_dataset_settings.py]:
```suggestion
        current_user: User = Security(auth.get_current_user),
```

File: src/argilla/server/apis/v0/handlers/text_classification_dataset_settings.py [hunk 4]
```diff
@@ -82,7 +82,7 @@ async def save_settings(
         ws_params: CommonTaskHandlerDependencies = Depends(),
         datasets: DatasetsService = Depends(DatasetsService.get_instance),
         validator: DatasetValidator = Depends(DatasetValidator.get_instance),
-        user: User = Security(auth.get_user, scopes=["write:dataset.settings"]),
+        user: User = Security(auth.get_current_user),
```
Comment (jfcalvo) [inline src/argilla/server/apis/v0/handlers/text_classification_dataset_settings.py]:
```suggestion
        current_user: User = Security(auth.get_current_user),
```

File: src/argilla/server/apis/v0/handlers/token_classification_dataset_settings.py [hunk 5]
```diff
@@ -54,7 +54,7 @@ async def get_dataset_settings(
         name: str = DATASET_NAME_PATH_PARAM,
         ws_params: CommonTaskHandlerDependencies = Depends(),
         datasets: DatasetsService = Depends(DatasetsService.get_instance),
-        user: User = Security(auth.get_user, scopes=["read:dataset.settings"]),
+        user: User = Security(auth.get_current_user),
```
Comment (jfcalvo) [inline src/argilla/server/apis/v0/handlers/token_classification_dataset_settings.py]:
```suggestion
        current_user: User = Security(auth.get_current_user),
```

File: src/argilla/server/apis/v0/handlers/token_classification_dataset_settings.py [hunk 6]
```diff
@@ -82,7 +82,7 @@ async def save_settings(
         ws_params: CommonTaskHandlerDependencies = Depends(),
         datasets: DatasetsService = Depends(DatasetsService.get_instance),
         validator: DatasetValidator = Depends(DatasetValidator.get_instance),
-        user: User = Security(auth.get_user, scopes=["write:dataset.settings"]),
+        user: User = Security(auth.get_current_user),
```
Comment (jfcalvo) [inline src/argilla/server/apis/v0/handlers/token_classification_dataset_settings.py]:
```suggestion
        current_user: User = Security(auth.get_current_user),
```

File: src/argilla/server/apis/v0/handlers/token_classification_dataset_settings.py [hunk 7]
```diff
@@ -110,7 +110,7 @@ async def delete_settings(
         name: str = DATASET_NAME_PATH_PARAM,
         ws_params: CommonTaskHandlerDependencies = Depends(),
         datasets: DatasetsService = Depends(DatasetsService.get_instance),
-        user: User = Security(auth.get_user, scopes=["delete:dataset.settings"]),
+        user: User = Security(auth.get_current_user),
```
Comment (jfcalvo) [inline src/argilla/server/apis/v0/handlers/token_classification_dataset_settings.py]:
```suggestion
        current_user: User = Security(auth.get_current_user),
```

File: src/argilla/server/policies.py [hunk 8]
```diff
@@ -66,9 +66,41 @@ def delete(cls, user: User) -> PolicyAction:
...
+        return lambda actor: actor.is_admin or dataset.workspace in [ws.name for ws in actor.workspaces]
+
+    @classmethod
+    def create(cls, user: User) -> bool:
+        return user.is_admin
+
+    @classmethod
+    def update(cls, dataset: Dataset) -> PolicyAction:
+        is_get_allowed = cls.get(dataset)
+        return lambda actor: actor.is_admin or (is_get_allowed(actor) and actor.username == dataset.created_by)
```
Comment (jfcalvo) [inline src/argilla/server/policies.py:84]: I believe we need to reduce complexity here someway. Let's discuss it.

Apart from that now that we have a ORM User instance on all handlers we can change all the policies from `actor.role == UserRole.admin` to `actor.is_admin`

**Review by jfcalvo — CHANGES_REQUESTED** [review]

File: src/argilla/server/policies.py (hunk 8 above)
Comment (frascuchon) [reply src/argilla/server/policies.py:84]: Yes, the way policy actions are compounded is not quite simple.

**Review by jfcalvo — APPROVED** [review]

**Pull request merged**